src/util/spanparsing.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_SPANPARSING_H
+#define BITCOIN_UTIL_SPANPARSING_H
+
+#include <string>
+#include <span.h>
+
+/** Parse a constant. If successful, sp is updated to skip the constant and return true. */
+bool Const(const std::string& str, Span<const char>& sp);
```

Comment:
with C++ throwing everything into the global namespace by default, I think these functions names  (`Const`, `Func`, `Expr`) are too general

Reply:
Make it a `namespace span` :) ?

Reply:
I've moved them into a namespace spanparsing, and added some documentation.